modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -26,31 +27,69 @@ function waitForJolokia() {
 
 endpointsCode=$(curl -s -o /dev/null -w "%{http_code}" -G -k -H "${endpointsAuth}" ${endpointsUrl})
 if [ $endpointsCode -ne 200 ]; then
-  echo "Can't find endpoints with ips status <${endpointsCode}>" 
+  echo "Can't find endpoints with ips status <${endpointsCode}>"
   exit 1
 fi
 
 ENDPOINTS=$(curl -s -X GET -G -k -H "${endpointsAuth}" ${endpointsUrl}"endpoints/${ENDPOINT_NAME}")
+echo "=====got endpoints details:"
 echo $ENDPOINTS
+# we will find out a broker pod's fqdn name which is <pod-name>.<$HEADLESS_SVC_NAME>.<namespace>.svc.<domain-name>
+# https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/
 count=0
+foundTarget="false"
 while [ 1 ]; do
   ip=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['ip']")
   if [ $? -ne 0 ]; then
     echo "Can't find ip to scale down to tried ${count} ips"
-    exit
+    exit 1
   fi
-
   echo "got ip ${ip} broker ip is ${BROKER_IP}"
   if [ "$ip" != "$BROKER_IP" ]; then
+    # found one, now get pod name
```

Comment:
the comment is opposite of the condition above. Shouldn't it be something like "Haven't found one, let's try through pod name" ?

Reply:
No the comment is correct. The aim here is to find a living pod that is not the "dead one". 
But maybe there is a problem with BROKER_IP comparison... (it is got from hostname -f which shouldn't be an ip address...)

Reply:
So there should be maybe another comment, that would reflect that. This way the condition looks like 
"I got an IP address, but that IP address must not be IP address of broker" (in another words, I am searching for IP of anything else but broker). It's a bit confusing. Naming convention for variables might be at fault here, so comment could clarify that.

Reply:
yeah let me make it more clear. thanks